include/beast/websocket/impl/write.ipp

```diff
@@ -0,0 +1,698 @@
+//
+// Copyright (c) 2013-2016 Vinnie Falco (vinnie dot falco at gmail dot com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+
+#ifndef BEAST_WEBSOCKET_IMPL_WRITE_IPP
+#define BEAST_WEBSOCKET_IMPL_WRITE_IPP
+
+#include <beast/core/bind_handler.hpp>
+#include <beast/core/buffer_cat.hpp>
+#include <beast/core/buffer_concepts.hpp>
+#include <beast/core/consuming_buffers.hpp>
+#include <beast/core/handler_alloc.hpp>
+#include <beast/core/prepare_buffers.hpp>
+#include <beast/core/static_streambuf.hpp>
+#include <beast/core/stream_concepts.hpp>
+#include <beast/core/detail/clamp.hpp>
+#include <beast/websocket/detail/frame.hpp>
+#include <boost/assert.hpp>
+#include <algorithm>
+#include <memory>
+
+namespace beast {
+namespace websocket {
+
+/*
+    template<class ConstBufferSequence>
+    void
+    write_frame(bool fin, ConstBufferSequence const& buffer)
+
+    Depending on the settings of autofragment role, and compression,
+    different algorithms are used.
+
+    1.  autofragment: false
+        compression:  false
+
+        In the server role, this will send a single frame in one
+        system call, by concatenating the frame header and the payload.
+
+        In the client role, this will send a single frame in one system
+        calls, using the write buffer to calculate masked data.
```

Comment:
All of the implementation notes at the top of write.ipp are removed in a later branch:
https://github.com/vinniefalco/Beast/blob/pmd/include/beast/websocket/impl/write.ipp